modules/engines/engine-xsodata/src/main/java/com/sap/xsk/xsodata/ds/handler/XSKScriptingOData2EventHandler.java

```diff
@@ -0,0 +1,501 @@
+      try {
+        closeConnection(connectionParam);
+        dropTemporaryTable((String) context.get(BEFORE_TABLE_NAME));
+      } catch (SQLException e) {
+        logger.error(e.getMessage(), e);
+        throw new InvalidStateException(e);
+      }
+    }
+  }
+
+  @Override
+  public void afterDeleteEntity(DeleteUriInfo uriInfo, String contentType, Map<Object, Object> context) {
+    Connection connectionParam = null;
+    try {
+      connectionParam = dataSource.getConnection();
+
+      context.put(CONNECTION, connectionParam);
+      context.put(BEFORE_TABLE_NAME, context.get(BEFORE_DELETE_ENTITY_TABLE_NAME));
+
+      super.afterDeleteEntity(uriInfo, contentType, context);
+    } catch (Exception e) {
+      logger.error(e.getMessage(), e);
+      throw new InvalidStateException(e);
+    } finally {
+      try {
+        closeConnection(connectionParam);
+        dropTemporaryTable((String) context.get(BEFORE_TABLE_NAME));
+      } catch (SQLException e) {
+        logger.error(e.getMessage(), e);
+        throw new InvalidStateException(e);
+      }
+    }
+  }
+
+  @Override
+  public ODataResponse onDeleteEntity(DeleteUriInfo uriInfo, String contentType, Map<Object, Object> context) {
+    SQLSelectBuilder selectBuilder = (SQLSelectBuilder) context.get(SELECT_BUILDER);
+    SQLContext sqlContext = (SQLContext) context.get(SQL_CONTEXT);
+
+    Connection connectionParam = null;
+    try {
+      connectionParam = dataSource.getConnection();
+
+      String beforeTableName = generateTemporaryTableName(uriInfo.getTargetType().getName());
+      createTemporaryTableAsSelect(connectionParam, beforeTableName, selectBuilder, sqlContext);
+
+      context.put(CONNECTION, connectionParam);
+      context.put(BEFORE_TABLE_NAME, beforeTableName);
+
+      return super.onDeleteEntity(uriInfo, contentType, context);
+    } catch (Exception e) {
+      logger.error(e.getMessage(), e);
+      throw new InvalidStateException(e);
+    } finally {
+      try {
+        closeConnection(connectionParam);
+        dropTemporaryTable((String) context.get(BEFORE_TABLE_NAME));
+      } catch (SQLException e) {
+        logger.error(e.getMessage(), e);
+        throw new InvalidStateException(e);
+      }
+    }
+  }
+
+  @Override
+  public String getName() {
+    return ODATA2_EVENT_HANDLER_NAME;
+  }
+
+  private String getSQLInsertBuilderTargetTable(SQLInsertBuilder insertBuilder, SQLContext sqlContext) throws ODataException {
+    SQLStatement sqlStatement = insertBuilder.build(sqlContext);
+    sqlStatement.sql();
+    return insertBuilder.getTargetTableName();
+  }
+
+  private void createTemporaryTableLikeTable(Connection connection, String temporaryTableName, String likeTableName) throws SQLException {
+    DBMetadataUtil dbMetadataUtil = new DBMetadataUtil();
+    String normalizedTableName = DBMetadataUtil.normalizeTableName(likeTableName);
+    String odataArtifactTypeSchema = dbMetadataUtil.getOdataArtifactTypeSchema(normalizedTableName);
```

Comment:
for this you should use XSKTableMetadataProvider because the dbMetadata util is returning the first schema that have artefact with that name which is wrong.